storage/innobase/clone/clone0repl.cc

```diff
@@ -459,6 +455,10 @@ void Clone_persist_gtid::flush_gtids(THD *thd) {
     clone operation. */
     clone_update_gtid_status(all_gtids);
     my_free(gtid_buffer);
+
+    /* Compress always if in recovery. */
+    compress_recovery = true;
```

Comment:
This is now independent of recovery mode. Is this intentional?

Before your patch, "!m_thread_active.load()" (which is also stored in is_recovery) is used to set compress_recovery only on recovery. Now it is unconditional

If this is a mistake, I would sugged to move this into the above if (is_recovery) code.

Reply:
Hi Satya,

Two things here.
1. We are setting compress_recovery fron inside the `if (is_recovery)` block only. 
2. This change is intentional and necessary to make the compression of the gtid_executed table even when InnoDB recovered zero GTIDs from the undo log post recovery. Right now it is compressed post recovery only when `m_num_gtid_mem.load() != 0` meaning only when InnoDB read at least one GTID from the undolog.

With this patch, we ensure that compression is done either by the persister or compressor threads on server restart.

Reply:
1) my bad, yeah.. it is already inside is_recovery block

2) why do we need compression of gtids from InnoDB if InnoDB doesn't any gtids from its undo header. Wouldn't it be better if gtid compressor thread do it on restart?

Reply:
Referring to https://dev.mysql.com/doc/refman/8.0/en/replication-gtids-concepts.html.

```When the server instance is started and the thread/sql/compress_gtid_table thread is launched, in most server configurations, compression is performed for the mysql.gtid_executed table. In releases before MySQL 8.0.17 when binary logging is enabled, compression is triggered by the fact of the binary log being rotated at startup. In releases from MySQL 8.0.20, compression is triggered by the thread launch. In the intervening releases, compression does not take place at startup.```

I probably didn't read it properly and was under the assumption that compression is not taking place even now when binary logs are rotated at startup. So I went with this approach to perform compression during server restart.

But this issue has been fixed by Oracle in 8.0.20 in a different way by setting the `should_compress` flag before entering the wait loop(https://github.com/mysql/mysql-server/blob/8.0/sql/rpl_gtid_persist.cc#L759).

So, my change is not required and will revert it as we are anyway merging the Oracle bugfix in our 8.0.20 merge.

Reply:
Good find!